src/fstab-generator/fstab-generator.c

```diff
@@ -1164,4 +1230,22 @@ static int run(const char *dest, const char *dest_early, const char *dest_late)
         return r < 0 ? r : r2 < 0 ? r2 : r3;
 }
 
-DEFINE_MAIN_GENERATOR_FUNCTION(run);
+static int run(int argc, char **argv) {
+        arg_sysroot_only = streq_ptr(argv[1], "--sysroot-only");
+
+        char **args = strv_skip(argv, 1 + arg_sysroot_only);
+
+        log_setup_generator();
```

Comment:
This should be called only when running as a generator. Otherwise, `log_setup()` should be used.

Reply:
Updated.